src/high-level/types/single-float.lisp

```diff
@@ -22,6 +22,37 @@
   matrix/single-float
   vector/single-float)
 
+;; Might want to inline this
+(defun %scalar= (s1 s2 epsilon)
+  "For equality checks of inexact scalars."
+  (declare (type number s1 s2) ; you can use this on complex but it might be slower because of the sqrt and two multiplies. Maybe.
+           (type real epsilon))
+  (<= (abs (- s1
+              s2))
+      epsilon))
+
+(defmethod =-lisp ((scalar1 rational) (scalar2 rational) &optional epsilon)
+  (declare (ignore epsilon))
+   "Rationals (integers and ratios) should be compared exactly."
+  (common-lisp:= scalar1 scalar2))
+
+(defmethod =-lisp ((scalar1 single-float) (scalar2 single-float) &optional (epsilon *float-comparison-threshold*))
+  (%scalar= scalar1 scalar2 epsilon))
+
+(defmethod =-lisp ((scalar1 rational) (scalar2 single-float) &optional (epsilon *float-comparison-threshold*))
+  (%scalar= scalar1 scalar2 epsilon))
+
+(defmethod =-lisp ((scalar1 single-float) (scalar2 rational) &optional (epsilon *float-comparison-threshold*))
+  (%scalar= scalar1 scalar2 epsilon))
+
+(defmethod =-lisp ((scalar1 float) (scalar2 single-float) &optional (epsilon *float-comparison-threshold*))
+  "This covers comparing double-float to single-float. Use least precise epsilon."
+  (%scalar= scalar1 scalar2 epsilon))
+
+(defmethod =-lisp ((scalar1 single-float) (scalar2 float) &optional (epsilon *float-comparison-threshold*))
+  "This covers comparing single-float to double-float. Use least precise epsilon."
+  (%scalar= scalar1 scalar2 epsilon))
+
```

Comment:
If we make the separate file, could we "just" implement a single `defmethod` on `real` and `real` to cover all these cases not covered by `rational`?